app.py

```diff
@@ -33,16 +44,24 @@ class User(Base):
 
     @classmethod
     def new(cls, username=None, password=None, session=DBSession):
+        """Stores password in database already hashed"""
         manager = BCRYPTPasswordManager()
         if not (username and password):
             raise ValueError("Username and password needed")
-        hashed = manager.encode(password)
-        instance = cls(username=username, password=hashed)
-        session.add(instance)
-        session.flush()
+        hashed = unicode(manager.encode(password))
+        try:
+            instance = cls(username=username, password=hashed)
+            session.add(instance)
+            session.flush()
+        except IntegrityError:
+            raise ValueError("Username already in use")
         return instance
 
     @classmethod
+    def all(cls, session=DBSession):
```

Comment:
Passing `DBSession` as a default value to this method definition feels squicky to me.  I cannot put my finger on exactly why, but given that `DBSession` is supposed to provide thread-local access to the database, setting it as a default feels to me like it should have bad results.